File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
from collections import deque


def solution(cards1, cards2, goal):
    answer = 'Yes'
    card1Queue = deque(cards1)
    card2Queue = deque(cards2)

    for i in goal:
        if len(card1Queue) > 0 and card1Queue[0] == i:
            card1Queue.popleft()
        elif len(card2Queue) > 0 and card2Queue[0] == i:
            card2Queue.popleft()
        else:
            answer = 'No'
            break

    return answer

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
import java.util.*;

class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        String answer = "Yes";
        
        Queue<String> card1Queue = new LinkedList<>(Arrays.asList(cards1));
        Queue<String> card2Queue = new LinkedList<>(Arrays.asList(cards2));

        for (int i=0; i<goal.length; i++){
            if (!card1Queue.isEmpty() && card1Queue.peek().equals(goal[i])) {
                card1Queue.poll();
            } else if (!card2Queue.isEmpty() && card2Queue.peek().equals(goal[i])) {
                card2Queue.poll();
            } else {
                answer = "No";
                break;
            }
        }
        
        return answer;
    }
}